software/controller/lib/hal/pwm.h

```diff
@@ -63,16 +42,24 @@ class PWM {
   // part in 4000 (80000000/20000) or about 12 bits.
   Frequency pwm_freq_;
 
-  TimerReg* timer_{nullptr};
-  uint8_t channel_{0};
-
+  TimerReg* timer_;
+  uint8_t channel_;
 #if !defined(BARE_STM32)
+  float value_{0};
```

Comment:
Here, it might also be time to move this out into a separate mock class, if not now then in next PR.

Reply:
left a TODO, on which I will focus my next efforts (as all those "mock" todos)